### PR TITLE
Fix: ListObjectsV2 behaving differently than AWS API 

### PR DESCRIPTION
**Pull request by irahulranjan**
https://github.com/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22+language%3Ago

### CODE REVIEW

File: tests/test_s3/test_s3.py [hunk 1]
```diff
@@ -1809,6 +1809,29 @@ def test_boto3_list_objects_v2_common_prefix_pagination():
...
+        "Bucket": "mybucket",
+        "Delimiter": "/",
+        "Prefix": "test/",
+        "MaxKeys": max_keys,
+        "ContinuationToken": "",
+    }
+
+    with pytest.raises(botocore.exceptions.ClientError) as exc:
+        s3.list_objects_v2(**args)
+    exc.value.response["Error"]["Code"].should.equal("InvalidArgument")
```
Comment (bblommers) [inline tests/test_s3/test_s3.py:1832]: Can we have another assertion that the `exc.value.response["Error"]["Message"]` is also set correctly? Happy to merge afterwards

**Review by bblommers — CHANGES_REQUESTED** [review]

File: tests/test_s3/test_s3.py (hunk 1 above)
Comment (irahulranjan) [reply tests/test_s3/test_s3.py:1832]: sure.

**Review by bblommers — APPROVED** [review]
LGTM. Thanks for the PR @irahulranjan!

**Comment by bblommers** [comment]
This is now part of moto >= 1.3.16.dev180

**Pull request merged**